src/smd-core.c

```diff
@@ -143,7 +143,7 @@ static int smd_attr_copy_val_to_internal(char *out, smd_dtype_t *t, smd_dtype_t
         case (SMD_TYPE_INT16): {
           int16_t ov = *(int16_t *)val;
           // check the range
-          if (ov <= INT8_MIN || ov >= INT8_MAX) {
+          if (ov <= INT8_MIN || ov > INT8_MAX) {
```

Comment:
Why didn't you remove the `=` from the `<=` as well? You do not want to exclude the value `-128` from the range, do you?